docs/pipelines.md

```diff
@@ -282,6 +284,28 @@ In this example, the task "build-the-image" will be executed and if the first
 run fails a second one would triggered. But, if that fails no more would
 triggered: a max of two executions.
 
+
+#### conditions
+
+Sometimes you will need to run tasks only when some conditions are true. The `conditions` field 
+allows you to list a series of references to [`Conditions`](#conditions) that are run before the task
+is run. If all of the conditions evaluate to true, the task is run. If any of the conditions are false,
+the Task is not run. Its status.ConditionSucceeded is set to False with the reason set to  `ConditionCheckFailed`.
+However, unlike regular task failures, condition failures do not automatically fail the entire pipeline 
+run -- other tasks in the pipeline are still run.
```

Comment:
might be good to state what happens here to subsequent Tasks that are `runAfter` or `from` this Task (I would assume that if a Task is using a resource `from` this Task, it couldn't possibly run? not as clear what the `runAfter` behaviour should be...)

Reply:
and i noticed that that case isn't present in the examples or the end to end tests - would probably be good to include somewhere! (or maybe in a follow up PR? but even in that case maybe we could document what our intended behaviour is)

Reply:
clarified that this is only for tasks that are not dependent. 
I think for `runAfter` we should be explicit and let the user select what to do. I have #1023 open for that.

Reply:
kk sounds good!